### PR TITLE
refactor(store): add useStore function and rewrite useShareStore

### PR DESCRIPTION
**Pull request by gene9831**
English | [简体中文](https://github.com/opentiny/tiny-engine/blob/develop/.github/PULL_REQUEST_TEMPLATE/PULL_REQUEST_TEMPLATE.zh-CN.md)

# PR

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our [Commit Message Guidelines](https://github.com/opentiny/tiny-engine/blob/develop/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Built its own designer, fully self-validated

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

### What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Summary by CodeRabbit

- **New Features**
	- Introduced `useState` and `useStore` hooks for managing state and reactive stores within the app.
  
- **Bug Fixes**
	- Ensured state consistency across components and handled non-existent keys appropriately.

- **Tests**
	- Added comprehensive test cases for `useState` and `useStore` hooks, including sharing state, patching stores, and converting objects to arrays.

### CODE REVIEW

File: packages/entry/test/useState.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    expect(state.value).toEqual('world 2')
+    expect(store).toEqual({ hello: 'world 2' })
+
+    patchStore({ hello: 'world 3' })
+
+    expect(state.value).toEqual('world 3')
+    expect(store).toEqual({ hello: 'world 3' })
+  })
+})
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/entry/test/useState.test.js]: > Test cases look comprehensive for the basic functionality of `useState` and `useStore`. Consider adding more edge cases, especially error handling scenarios.
> 
> Would you like me to help draft additional test cases covering edge scenarios?

@coderabbitai Yes, help me

File: packages/utils/src/utils/index.js [hunk 2]
```diff
@@ -325,3 +325,23 @@ function _deepClone(target, map) {
...
+
+  const arr = Object.entries(obj)
+    // 过滤掉非对象的值，防止后面展开对象失败
+    .filter(([, value]) => typeof value === 'object' && obj !== null)
+    .map(([key, value]) => ({ ...value, [keyTo]: key }))
+    .map(({ [orderKey]: order, ...rest }) => ({ ...rest, [orderKey]: order ?? Number.MAX_SAFE_INTEGER }))
+    .sort((a, b) => a[orderKey] - b[orderKey])
+
+  return arr
+}
```
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/utils/src/utils/index.js:347]: `keyTo` and `orderKey` is optional

File: packages/entry/test/useState.test.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (gene9831) [reply packages/entry/test/useState.test.js]: @coderabbitai The `useState` I coded has bit different from React's. The first parameter of my `useState` is a key, is not a initial value. The reason why I implement is this way is to share state globally. So the above tests about `useState` is impassable for me.

**Review by hexqi — DISMISSED** [review]

**Review by chilingling — APPROVED** [review]

**Review by hexqi — APPROVED** [review]
